tests/metrics/run.sh

```diff
@@ -0,0 +1,116 @@
+	run_sql_source2 "alter table metrics.t2 add column new_col1 int;"
+	# test dml lag metric >= 1 beacuse we inject updateReplicationLag(ddl) to sleep(1)
+	check_metric $WORKER1_PORT "dm_syncer_replication_lag{task=\"test\"}" 5 0 999
+	check_metric $WORKER2_PORT "dm_syncer_replication_lag{task=\"test\"}" 5 0 999
+	# check two worker's secondsBehindMaster > 0
+	check_secondsBehindMaster 0 2
+	echo "check ddl done!"
+
+	# restart dm worker
+	kill_dm_worker
+	export GO_FAILPOINTS="github.com/pingcap/dm/syncer/BlockSyncerUpdateLag=return(\"insert,2\");github.com/pingcap/dm/syncer/changeTickerInterval=return(10)"
+	run_dm_worker $WORK_DIR/worker1 $WORKER1_PORT $cur/conf/dm-worker1.toml
+	check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER1_PORT
+	run_dm_worker $WORK_DIR/worker2 $WORKER2_PORT $cur/conf/dm-worker2.toml
+	check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER2_PORT
+
+	run_sql_source1 "truncate table metrics.t1;"                                        # make skip job
+	run_sql_file $cur/data/db1.increment.sql $MYSQL_HOST1 $MYSQL_PORT1 $MYSQL_PASSWORD1 # make dml job
+	run_sql_file $cur/data/db2.increment.sql $MYSQL_HOST2 $MYSQL_PORT2 $MYSQL_PASSWORD2 # make dml job
+
+	# test dml lag metric >= 2 beacuse we inject updateReplicationLag(insert) to sleep(2)
+	# although skip lag is 0 (locally), but we use that lag of all dml/skip lag, so lag still >= 2
+	check_metric $WORKER1_PORT "dm_syncer_replication_lag{task=\"test\"}" 5 1 999
+	check_metric $WORKER2_PORT "dm_syncer_replication_lag{task=\"test\"}" 5 1 999
+	check_secondsBehindMaster 1 2
+	echo "check dml/skip done!"
+
+	# restart dm worker
+	kill_dm_worker
+	export GO_FAILPOINTS=''
+	run_dm_worker $WORK_DIR/worker1 $WORKER1_PORT $cur/conf/dm-worker1.toml
+	check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER1_PORT
+	run_dm_worker $WORK_DIR/worker2 $WORKER2_PORT $cur/conf/dm-worker2.toml
+	check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER2_PORT
+	check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
+	# check the dmctl query-status no new dml, lag should be set to 0
+	run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"query-status test" \
+		"\"secondsBehindMaster\": \"0\"" 2
+	echo "check zero job done!"
+
+	# check the time interval between dm-worker updates when the queue is empty, the desired time interval is 1s
+	kill_dm_worker
+	export GO_FAILPOINTS="github.com/pingcap/dm/syncer/changeTickerInterval=return(1)"
+	run_dm_worker $WORK_DIR/worker1 $WORKER1_PORT $cur/conf/dm-worker1.toml
+	check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER1_PORT
+	run_dm_worker $WORK_DIR/worker2 $WORKER2_PORT $cur/conf/dm-worker2.toml
+	check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER2_PORT
+	sleep 3 # we need sleep 3 because current ticker interval is 1 and we need at least 2 ticker's log
+	$cur/../_utils/check_ticker_interval.py $WORK_DIR/worker1/log/dm-worker.log 1
```

Comment:
Why do we check interval here?

Reply:
to make sure ticker Reset is working properly
---
resets the time interval for each loop to prevent a certain amount of time being spent on the previous ticker execution to `executeSQLs` resulting in the next tikcer not waiting for the full waitTime.

Reply:
But I don't think this test can make sure "the next ticker not waiting for the full waitTime". If you set `changeTickerInterval` to 1 this value will definitely be larger than 1. What's more, there is no new job except `flushJob` from `jobChan`, so this also can't satisfy your expectations.

Reply:
> But I don't think this test can make sure "the next ticker not waiting for the full waitTime". If you set `changeTickerInterval` to 1 this value will definitely be larger than 1. What's more, there is no new job except `flushJob` from `jobChan`, so this also can't satisfy your expectations.

originally there's a bug 😂 so not "...definitely be larger than 1"

Reply:
I know that bug... I mean this test can't make us confirm that bug is fixed.